jenkins-integration/beaker/install/shared/r10k_deploy.rb

```diff
@@ -15,21 +18,65 @@ def install_r10k(host)
   on(host, "#{gem} install r10k --no-document")
 end
 
-def create_r10k_config(host, r10k)
-  configfile = r10k_configpath(r10k)
-  configdir = '/etc/puppetlabs/r10k'
-  on(host, "mkdir -p #{configdir}")
-  scp_to(host, configfile, "#{configdir}/r10k.yaml")
+def create_r10k_config(host, r10k_config)
+
+  on(host, "mkdir -p #{R10K_DIR}")
+
+  cachedir = "#{R10K_DIR}/cache"
+
+  r10k_config_contents = <<EOS
+:cachedir: '#{cachedir}'
+
+:sources:
+  :perf-test:
+    remote: '#{r10k_config[:control_repo]}'
+    basedir: '#{r10k_config[:basedir]}'
+EOS
+
+  create_remote_file(host, R10K_CONFIG_PATH, r10k_config_contents)
 end
 
-def run_r10k_deploy(host)
+def run_r10k_deploy(host, r10k_config)
   r10k = '/opt/puppetlabs/puppet/bin/r10k'
-  on(host, "#{r10k} deploy environment --puppetfile --verbose")
+  r10k_config[:environments].each do |env|
+    on(host, "#{r10k} deploy environment #{env} -p -v debug -c #{R10K_CONFIG_PATH}")
+  end
+end
+
+
+########################
+
+step "Install git" do
+  on(master, puppet_resource("package git ensure=installed"))
+end
+
+step "Set up SSH key for github access" do
+  if !master.file_exist?("/root/.ssh/id_rsa")
+    result = curl_on master, "-o /root/.ssh/id_rsa 'http://int-resources.ops.puppetlabs.net/QE%20Shared%20Resources/travis_keys/id_rsa'"
+    assert_equal 0, result.exit_code
+
+    on(master, "chmod 600 /root/.ssh/id_rsa")
+
+    result = curl_on master, "-o /root/.ssh/id_rsa.pub 'http://int-resources.ops.puppetlabs.net/QE%20Shared%20Resources/travis_keys/id_rsa.pub'"
+    assert_equal 0, result.exit_code
+  end
+end
+
+step "add github to known hosts" do
+  # Create known_hosts file with GitHub host key to prevent
+  # "Host key verification failed" errors during clones
+  create_remote_file(master, "/root/.ssh/known_hosts", <<-EOS)
```

Comment:
Is this the only known host that the process needs?  Just wondering if we should append to whatever might already be there rather than just replacing it entirely?